# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IDBIndex.cpp
# Commit: 0e77abe58616
# Full Hash: 0e77abe58616924db5bf2f2a5ca2f7c30f3031e1
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-09-11 21:53:06
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Change definition of OpenCursorParams subtypes to allow reducing duplicated code r=ttung,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D40955
# ==============================================================================

diff -r 556059030dc6 -r 0e77abe58616 dom/indexedDB/IDBIndex.cpp
--- a/dom/indexedDB/IDBIndex.cpp	Wed Sep 11 11:51:28 2019 +0000
+++ b/dom/indexedDB/IDBIndex.cpp	Wed Sep 11 11:51:26 2019 +0000
@@ -458,18 +458,22 @@
   OpenCursorParams params;
   if (aKeysOnly) {
     IndexOpenKeyCursorParams openParams;
-    openParams.objectStoreId() = objectStoreId;
-    openParams.indexId() = indexId;
-    openParams.optionalKeyRange() = std::move(optionalKeyRange);
-    openParams.direction() = direction;
+    openParams.commonIndexParams().commonParams().objectStoreId() =
+        objectStoreId;
+    openParams.commonIndexParams().indexId() = indexId;
+    openParams.commonIndexParams().commonParams().optionalKeyRange() =
+        std::move(optionalKeyRange);
+    openParams.commonIndexParams().commonParams().direction() = direction;
 
     params = std::move(openParams);
   } else {
     IndexOpenCursorParams openParams;
-    openParams.objectStoreId() = objectStoreId;
-    openParams.indexId() = indexId;
-    openParams.optionalKeyRange() = std::move(optionalKeyRange);
-    openParams.direction() = direction;
+    openParams.commonIndexParams().commonParams().objectStoreId() =
+        objectStoreId;
+    openParams.commonIndexParams().indexId() = indexId;
+    openParams.commonIndexParams().commonParams().optionalKeyRange() =
+        std::move(optionalKeyRange);
+    openParams.commonIndexParams().commonParams().direction() = direction;
 
     params = std::move(openParams);
   }